### ISSUE TITLE
gif image issue

### ISSUE THREAD

**Issue by sumitpaytm** [SDWebImage/SDWebImage#2818]
### New Issue Checklist
gif image is not working for below pod 
pod 'SDWebImage/WebP', '4.4.4'

if I use pod 'SDWebImage/WebP', '3.8.0', then it is working fine which has below method:
- (id <SDWebImageOperation>)downloadImageWithURL:(NSURL *)url
                                         options:(SDWebImageOptions)options
                                        progress:(SDWebImageDownloaderProgressBlock)progressBlock
                                       completed:(SDWebImageCompletionWithFinishedBlock)completedBlock;

in pod 'SDWebImage/WebP', '4.4.4', This method has changed into below method which is not working for gif image.
 (nullable id <SDWebImageOperation>)loadImageWithURL:(nullable NSURL *)url
                                              options:(SDWebImageOptions)options
                                             progress:(nullable SDWebImageDownloaderProgressBlock)progressBlock
                                            completed:(nullable SDInternalCompletionBlock)completedBlock;

* [ ] I have read and understood the [CONTRIBUTING guide](https://github.com/rs/SDWebImage/blob/master/.github/CONTRIBUTING.md)
* [ ] I have read the [Documentation](http://cocoadocs.org/docsets/SDWebImage/)
* [ ] I have searched for a similar issue in the [project](https://github.com/rs/SDWebImage/issues) and found none

### Issue Info

 Info                    | Value                               |
-------------------------|-------------------------------------|
 Platform Name           | e.g. ios / macos / tvos / watchos
 Platform Version        | e.g. 12.0 / 10.14.0 / 12.0 / 5.0
 SDWebImage Version      | e.g. 5.0.0 / 4.4.0
 Integration Method      | e.g. carthage / cocoapods / manually
 Xcode Version           | e.g. Xcode 11 / Xcode 10
 Repro rate              | e.g. all the time (100%) / sometimes x% / only once
 Repro with our demo prj | e.g. does it happen with our demo project?
 Demo project link       | e.g. link to a demo project that highlights the issue

### Issue Description and Steps

Please fill in the detailed description of the issue (full output of any stack trace, compiler error, ...) and the steps to reproduce the issue.

**Comment by dreampiggy** [comment]
@sumitpaytm See readme in 4.x version. It already answer the question what you want.

https://github.com/SDWebImage/SDWebImage/tree/4.x#animated-images-gif-support

SDWebImage 4.0 drop the built in support for GIF animation playing on UIImageView, you should use FLAnimaterdImageView. In 4.1, you can enable this again by using `SDWebImageGIFCoder`, but this may effect the FLAnimatedImageView performance, it's a paird-off.


SDWebImage 5.0 introduce a SDAnimatedImageView and deprecated FLAnimatedImageView support (into another plugin repo). And by default we support both UIImageView/SDAnimatedImageView GIF animation playing, no extra performance cost is used.

**Comment by sumitpaytm** [comment]
@dreampiggy, Thank you for your quick response. It is highly appreciated. Actually I want to use all image format including **webp** as well as **gif** in a single imageview. so could you please suggest me that which SDWebImage version will best suit for me?

**Comment by dreampiggy** [comment]
@sumitpaytm Use 5.0 with SDAnimatedImageView is the better solution on performance for Animation...See more detail reason here of `Why I don't recommended UIImageView and FLAnimatedImage`: [Wiki - Animated Image](https://github.com/SDWebImage/SDWebImage/wiki/Advanced-Usage#animated-image-50)

**Comment by sumitpaytm** [comment]
@dreampiggy I think 5.0 has no **webp** support?

**Comment by kinarobin** [comment]
You should add:
```ruby
pod 'SDWebImageWebPCoder', '0.2.3'
```

```
[[SDImageCodersManager sharedManager] addCoder:[SDImageWebPCoder sharedCoder]];
```

**Comment by sumitpaytm** [comment]
@kinarobin By using  'SDWebImageWebPCoder', '0.2.3' . Can I use **webp** and **gif** both in simple imageview?

**Comment by kinarobin** [comment]
If you want to use gif :
```ruby
[[SDImageCodersManager sharedManager] addCoder:[SDImageGIFCoder sharedCoder]];
```

**Comment by sumitpaytm** [comment]
@dreampiggy, @kinarobin It is working now. Thank you very much for your support. It's really nice
